skimage/morphology/footprints.py

```diff
@@ -95,6 +98,107 @@ def footprint_from_sequence(footprints):
     return morphology.binary_dilation(imag, footprints)
 
 
+def footprint_rectangular(shape, *, dtype=np.uint8, decomposition=None):
+    """Generate a rectangular or hyper-rectangular footprint.
+
+    Generates, depending on the length and dimensions requested with `shape`,
+    a square, rectangle, cube, cuboid, or even higher-dimensional versions
+    of these shapes.
+
+    Parameters
+    ----------
+    shape : tuple[int, ...]
+        The length of the footprint in each dimension. The length of the
+        sequence determines the number of dimensions of the footprint.
+    dtype : data-type, optional
+        The data type of the footprint.
+    decomposition : {None, 'separable', 'sequence'}, optional
+        If None, a single array is returned. For 'sequence', a tuple of smaller
+        footprints is returned. Applying this series of smaller footprints will
+        give an identical result to a single, larger footprint, but often with
+        better computational performance. See Notes for more details.
+        With 'separable', this function uses separable 1D footprints for each
+        axis. Whether 'sequence' or 'separable' is computationally faster may
+        be architecture-dependent.
+
+    Returns
+    -------
+    footprint : array or tuple[tuple[ndarray, int], ...]
```

Comment:
```suggestion
    footprint : array or tuple[tuple[array, int], ...]
```
or `ndarray or tuple[tuple[array, int], ...]`?